sklearn/utils/__init__.py

```diff
@@ -79,6 +81,31 @@
 _IS_32BIT = 8 * struct.calcsize("P") == 32
 
 
+def in_unstable_openblas_configuration():
+    """Return True if in an unstable configuration for OpenBLAS.
+
+    See discussions in https://github.com/numpy/numpy/issues/19411
+    """
+
+    # Make sure numpy and scipy are imported
+    import numpy  # noqa
+    import scipy  # noqa
+
+    modules_info = threadpool_info()
+
+    open_blas_used = any(info["internal_api"] == "openblas" for info in modules_info)
+    if not open_blas_used:
+        return False
+
+    open_blas_version = next(
+        info["version"] for info in modules_info if info["internal_api"] == "openblas"
+    )
+
+    return platform.machine() == "neoversen1" and open_blas_version < LooseVersion(
+        "0.3.16"
+    )
```

Comment:
Use `.get` just in case the version or architecture cannot be detected (to avoid failing with older versions of openblas and threadpoolctl).

Also rely on threadpoolctl to detect whether we use the `neoversen1` kernel with OpenBLAS:

```suggestion
    good_openblas_version = LooseVersion("0.3.16")
    for info in modules_info:
        if info["internal_api"] != "openblas":
            continue
        openblas_version = info.get("version")
        openblas_architecture = info.get("architecture")
        if openblas_version is None or openblas_architecture is None:
            # Cannot be sure that OpenBLAS is good enough. Assume unstable:
            return True
        if (
            openblas_architecture == "neoversen1"
            and openblas_version < good_openblas_version
        ):
            return True
    return False
```